feat: added additional context to workflow core and extended browser-sdk

chesterkmr opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced `additionalContext` support in the StateManager, enhancing state management capabilities.
	- Added a new hook, `useAdditionalWorkflowContext`, for improved workflow context retrieval.
- **Bug Fixes**
	- Updated methods to ensure proper context handling during plugin invocation.
- **Documentation**
	- Changelog updated to reflect new features and dependency updates across various packages.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

chesterkmr committed:
feat: added additional context to workflow core and extended browser-sdk

tomer-shvadron commented on apps/kyb-app/src/pages/CollectionFlow/CollectionFlow.tsx:
```diff
@@ -132,6 +134,8 @@ export const CollectionFlow = withSessionProtected(() => {
   if (getCollectionFlowState(initialContext)?.status === CollectionFlowStatusesEnum.failed)
     return <FailedScreen />;
 
+  console.log('additionalContext', additionalContext);
```
Remove

tomer-shvadron commented on sdks/workflow-browser-sdk/package.json:
```diff
@@ -60,6 +60,7 @@
     "cross-fetch": "^3.1.5",
     "cspell": "^6.31.2",
     "cz-conventional-changelog": "^3.3.0",
+    "dompurify": "^3.0.6",
```
What do we need `dompurify` for?

chesterkmr committed:
fix: cleanup

chesterkmr committed:
Merge branch 'dev' into illiar/feat/workflow-core-additional-context

alonp99 committed:
feat(workflows-service): update dependencies and refine workflow logic

- Update common, workflow-core and workflow-node-sdk versions
- Remove unnecessary createCollectionFlowToken flag from workflows
- Enhance context creation for workflow runtime data with new logic

(your workflows are more complicated than a Rubik's cube in a tornado)
